src/server.h

```diff
@@ -35,6 +35,7 @@
 #include "solarisfixes.h"
 #include "rio.h"
 #include "commands.h"
+#include "allocator_defrag.h"
```

Comment:
It's only used by server.c and defrag.c, so is it better to make it visible only in those files?

Reply:
the HAVE_DEFRAG macro is also used in config.c, I'm fine with either, let me know if you think it's cleaner to explicitly include

Reply:
Slight preference to include it in each file, but there doesn't seem to be any consistent rule here. Leave it in server.h if you wish. :)

Is it dangerous to have `#ifdef HAVE_DEFRAG` when the macro is defined in an include file? If we forget to include it, the macro is not defined and we will use the wrong branch?

If we always define it to 0 or 1 and use `#if HAVE_DEFRAG` instead of `ifdef`, is it safer?

Reply:
I would just get rid of the HAVE_DEFRAG  macro and use runtime check if the defrag is supported. I think it will be required anyway if we want to enable defrag in test mode.

Reply:
OK, make sense, but we can leave that for a future PR. Let's get this one merged.

Reply:
once this is merged I'll open an issue so we can track it

Reply:
I guess it needs to be handled in #1303.